### ISSUE TITLE
knife vault update Issue-  Error: invalid option: -C

### ISSUE THREAD

**Issue by pariminaresh** [chef/chef-vault#249]
Trying to update clients of a vault item.But facing an error- Error: invalid option: -C.

From docs '-C' is used to update clients.Why is it not working for me?Any version updates I need to do?

Thanks,

**Comment by marzzz21** [comment]
I'm also experiencing this problem. Tried `knife vault update private_keys mirror_git_key -S "role:my_role_name" -A "admin1, admin2" -VV` but I get this error:
https://gist.github.com/marzzz21/572064d1ff477b46e27140ebcef0125f

**Comment by rneu31** [comment]
@pariminaresh if you do `knife vault update --help` does -C show up as an option? Guessing you're using an older version.

**Comment by pariminaresh** [comment]
@rneu31 I don't see '-C' option either in the options list, chef-client version I have is 12.15.19.

**Comment by youhyunkim** [comment]
+1. Seems like a lot of the global parameters are missing.
```
$ knife -v
Chef: 12.16.42
```

**Comment by thommay** [comment]
@marzzz21 that looks like a completely separate issue, please file a new bug.

**Comment by thommay** [comment]
Are you all using `knife vault` from the ChefDK? Can you run:
```
chef gem list chef-vault
```
to get the version of chef-vault you have installed? If you are not using ChefDK, please run `gem list chef-vault` instead.

**Comment by youhyunkim** [comment]
I'm using `knife vault` from ChefDK.

Here's both:
```
$ chef gem list chef-vault
*** LOCAL GEMS ***
chef-vault (2.9.0)
```
```
$ gem list chef-vault
*** LOCAL GEMS ***
chef-vault (2.9.0)
```

**Comment by kbrowns** [comment]
I'm seeing the same behavior - using chef-vault 2.9.0 from ChefDK and -C/--clients is missing

**Comment by victort** [comment]
:+1:

**Comment by kamaradclimber** [comment]
@victort, `-S` option is made to specify a search query (in the same format as `knife search`). All nodes matching this search query will have access to the secret.
`knife vault refresh` can be used to refresh the list of nodes matching the search query and encrypt the symetrical key for them.

**Comment by kamaradclimber** [comment]
`--clients` option has been introduced by daddf78 which was initially released in v3.0.0.rc1
